File: src/main/assembly/help/Tutorial_js/5_Utils/55_Interlocks.js
/////////////////////////////////////////////////////////////////////////////////////////////////// 
// Interlocks: example on creating and installing device interlock rules.
/////////////////////////////////////////////////////////////////////////////////////////////////// 

//Motor and Positioners
var MyInterlock1 = Java.extend(Interlock)
var interlock1 = new MyInterlock1( [m1, p1]) {	    	    
    check: function (pos) {
    	m=pos[0]
    	p=pos[1]
        if ((p<500) && (m<5) && (m>4)){
            return false
        }
        return true        	
    },
}


/*
//Motor group
var MyInterlock2 = Java.extend(Interlock)
var interlock2 = new MyInterlock2( [mg1, p1]) {	    	    
    check: function (pos) {
    	print(pos)
    	mg=pos[0]
    	p=pos[1]
    	print(to_array(mg))
        if ((p<500) && (mg[0]<5) && (mg[1]>4)){
            return false
        }
        return true        	
    },
}
*/

/*
//Discrete Positioner
var MyInterlock3 = Java.extend(Interlock)
interlock3 = new MyInterlock3( [dp1, p1]) {	    	    
    check: function (pos) {
    	dp=pos[0]
    	p=pos[1]
    	print ("DP " + dp)
    	print ("P " + p)
        if ((p<500) && (dp=="Out")){
            return false
        }
        return true        	
    },
}
*/

File: src/main/assembly/help/Tutorial_py/5_Utils/55_Interlocks.py
################################################################################################### 
# Interlocks: example on creating and installing device interlock rules.
################################################################################################### 


class MyInterlock1 (Interlock):
#Motor and Positioners
    def __init__(self):
        Interlock.__init__(self, (m1, p1))

    def check(self, (m, p)):
        if p<500 and (m<5 and m>4):
            return False
        return True

interlock1 = MyInterlock1()

"""
#Motor group
class MyInterlock2(Interlock):
    def __init__(self):
        Interlock.__init__(self, (mg1, p1))

    def check(self, ((m1,m2), p)):
        if p<500 and (m1>4 and m2>4):
            return False
        return True

interlock2 = MyInterlock2()
"""


"""
#Discrete Positioner
class MyInterlock3(Interlock):
    def __init__(self):
        Interlock.__init__(self,  (dp1, p1))

    def check(self, (dp, p)):
        if p<500 and dp=="Out":
            return False
        return True

#interlock3 = MyInterlock3()
"""